File: NSOperation(OC)/NSOperation(OC)/AROperation.m
//
//  AROperation.m
//  NSOperation(OC)
//
//  Created by andyron<http://andyron.com> on 2018/3/15.
//  Copyright © 2018年 andyron. All rights reserved.
//

#import "AROperation.h"

@implementation AROperation

- (void)main {
    if (!self.isCancelled) {
        for (int i = 0; i < 2; i++) {
            [NSThread sleepForTimeInterval:2];
            NSLog(@"自定义Operation---%@", [NSThread currentThread]);
        }
    }
}


@end

File: NSOperation(Swift)/NSOperation(Swift)/AROperation.swift
//
//  AROperation.swift
//  NSOperation(Swift)
//
//  Created by Andy Ron on 2018/8/26.
//  Copyright © 2018年 Andy Ron. All rights reserved.
//

import UIKit

class AROperation: Operation {

    override func main() {
        if !self.isCancelled {
            for _ in 0..<2 {
                Thread.sleep(forTimeInterval: 2)
                print("自定义Operation----\(Thread.current)")
            }
        }
    }
    
}
